logstash-core/lib/logstash/config/source/modules.rb

```diff
@@ -0,0 +1,44 @@
+# encoding: utf-8
+require "logstash/config/source/base"
+require "logstash/config/pipeline_config"
+require "logstash/util/loggable"
+require "logstash/errors"
+
+module LogStash module Config module Source
+  class Modules < Base
+    include LogStash::Util::Loggable
+    def pipeline_configs
+      pipelines = []
+      plugin_modules = LogStash::PLUGIN_REGISTRY.plugins_with_type(:modules)
+
+      modules_array = @settings.get("modules.config") == [] ? @settings.get("modules") : @settings.get("modules.config")
+      logger.debug("MODULES_ARRAY", :modules_array => modules_array.to_s)
+      module_names = []
+      modules_array.each {|module_hash| module_names << module_hash["name"]}
+      if module_names.length > module_names.uniq.length
+        raise LogStash::ConfigLoadingError, I18n.t("logstash.modules.configuration.modules-must-be-unique", :modules => module_names)
+      end
+      ### Here is where we can force the modules_array to use only [0] for 5.5, and leave
+      ### a warning/error message to that effect.
+      modules_array.each do |module_hash|
+        begin
+          thismodule = plugin_modules.find { |allmodules| allmodules.modulename == module_hash["name"] }
+          altname = "module-#{module_hash["name"]}"
+          pipeline_id = module_hash["pipeline_id"].nil? ? altname : module_hash["pipeline_id"]
```

Comment:
We shouldn't even let users specify `pipeline_id`, to simplify -- for 6.0 as well as 5.x. We can infer from the module name as you've already done (`altname`). One less config to worry about.